File: backend/app/schemas/molecule.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class MoleculeBase(BaseModel):
    name: str
    charge: int
    multiplicity: int
    structure_xyz: str
    bundle_id: int


class MoleculeCreate(MoleculeBase):
    pass


class MoleculeUpdate(BaseModel):
    name: Optional[str] = None
    charge: Optional[int] = None
    multiplicity: Optional[int] = None
    structure_xyz: Optional[str] = None


class MoleculeResponse(MoleculeBase):
    id: int
    latest_job_id: Optional[str]

    class Config:
        orm_mode = True

File: frontend/src/types/molecule.ts
export interface Molecule {
  id: number;
  name: string;
  charge: number;
  multiplicity: number;
  structure_xyz: string;
  bundle_id: number;
  latest_job_id?: string;
}

export interface MoleculeCreate {
  name: string;
  charge: number;
  multiplicity: number;
  structure_xyz: string;
  bundle_id: number;
}

export interface MoleculeUpdate {
  name?: string;
  charge?: number;
  multiplicity?: number;
  structure_xyz?: string;
}
